explorer/src/context/main.tsx

```diff
@@ -1,17 +1,33 @@
+import { PaletteMode } from '@mui/material';
 import * as React from 'react';
+import { GatewayResponse, MixNodeResponse, ValidatorsResponse } from 'src/typeDefs/node-status-api-client';
+// import { MixNodeResponse, GatewayResponse } from 'src/typeDefs/node-status-api-client';
 import { Api } from '../api';
 
-export const MainContext = React.createContext({});
+type NodeApiResponse = {
+  data: MixNodeResponse | GatewayResponse | ValidatorsResponse | number | null
+  error: string | null
+}
+interface State {
+  mode: PaletteMode
+  toggleMode: () => void
+  mixnodes: NodeApiResponse | null
+  gateways: NodeApiResponse | null
+  validators: NodeApiResponse | null
+  block: NodeApiResponse | null
```

Comment:
```suggestion
  mixnodes: MixNodeResponse | null
  gateways: GatewayResponse | null
  validators: ValidatorsResponse | null
  block: BlockResponse | null
```

Reply:
You may have been trying to do something like this:

```
interface ApiState<RESPONSE> {
  isLoading: boolean;
  response?: RESPONSE;
  error?: Error;
}

interface State {
  mode: PaletteMode
  toggleMode: () => void
  mixnodes?: ApiState<MixNodeResponse>,
  gateways?: ApiState<GatewayResponse>,
  validators?: ApiState<ValidatorsResponse>,
  block?: ApiState<BlockResponse>,
}
```

When `mixnode` is `undefined` it is "not preset" i.e. never loaded. When it starts loading, the value changes to `mixnode: { isLoading: true }`, when an error: `mixnode: { isLoading: false, error: { message: 'Oh no!' } }` and with data: `mixnode: { isLoading: false, response: {...}}`

Reply:
Thanks for this - i _think_ it's sorted now.